internal/config/node/cgroups.go

```diff
@@ -8,62 +8,33 @@ import (
 	"strings"
 	"sync"
 
-	libpodcgroups "github.com/containers/libpod/pkg/cgroups"
-	libctrcgroups "github.com/opencontainers/runc/libcontainer/cgroups"
+	"github.com/containers/libpod/pkg/cgroups"
 	"github.com/pkg/errors"
 )
 
 var (
-	cgroupHasHugetlbOnce sync.Once
-	cgroupHasHugetlb     bool
-	cgroupHasHugetlbErr  error
-
-	cgroupHasPidOnce sync.Once
-	cgroupHasPid     bool
-	cgroupHasPidErr  error
+	cgroupIsV2Once sync.Once
+	cgroupIsV2     bool
+	cgroupIsV2Err  error
 
 	cgroupHasMemorySwapOnce sync.Once
 	cgroupHasMemorySwap     bool
 	cgroupHasMemorySwapErr  error
 
-	cgroupIsV2Once sync.Once
-	cgroupIsV2     bool
-	cgroupIsV2Err  error
+	cgroupControllerOnce sync.Once
+	cgroupControllerErr  error
+	cgroupHasHugetlb     bool
+	cgroupHasPid         bool
 )
 
-// CgroupHasHugetlb returns whether the hugetlb controller is present
-func CgroupHasHugetlb() bool {
-	cgroupHasHugetlbOnce.Do(func() {
-		if CgroupIsV2() {
-			controllers, err := ioutil.ReadFile("/sys/fs/cgroup/cgroup.controllers")
-			if err != nil {
-				cgroupHasHugetlbErr = err
-				return
-			}
-			cgroupHasHugetlb = strings.Contains(string(controllers), "hugetlb")
-			return
-		}
-
-		st, err := os.Stat("/sys/fs/cgroup/hugetlb")
-		if err != nil {
-			cgroupHasHugetlbErr = err
-			return
-		}
-		if st.IsDir() {
-			cgroupHasHugetlb = true
-		}
-	})
-	return cgroupHasHugetlb
-}
-
-// CgroupHasPid returns whether the pid controller is present
-func CgroupHasPid() bool {
-	cgroupHasPidOnce.Do(func() {
-		_, err := libctrcgroups.FindCgroupMountpoint("", "pids")
-		cgroupHasPid = err == nil
-		cgroupHasPidErr = err
+// CgroupIsV2 returns whether we are using cgroup v2 or v1
+func CgroupIsV2() bool {
+	cgroupIsV2Once.Do(func() {
```

Comment:
You can just do

```go
func CgroupIsV2() bool {
  v2, _ := cgroups.IsCgroup2UnifiedMode()
  return v2
}
```

(and add error checking somewhere in init)

Reply:
I used libcontainer as suggested below